openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -344,29 +345,75 @@ def post(self, request):
         Marks the user as having no password set for deactivation purposes,
         and logs the user out.
         """
-        username = None
         user_model = get_user_model()
         try:
             # Get the username from the request and check that it exists
-            username = request.data['username']
-            user = user_model.objects.get(username=username)
-
+            verify_user_password_response = self._verify_user_password(request)
+            if verify_user_password_response.status_code != status.HTTP_204_NO_CONTENT:
+                return verify_user_password_response
             with transaction.atomic():
                 # 1. Unlink LMS social auth accounts
-                UserSocialAuth.objects.filter(user_id=user.id).delete()
+                UserSocialAuth.objects.filter(user_id=request.user.id).delete()
```

Comment:
Should / can we add logging so that for a rollback we know what the cause of the rollback was?

I know we handle a couple of error cases below, but I don't think we'd be able to distinguish if a rollback occurred due to being unable to add a user to a retirement queue or if we're unable to log a user out, for example.

Reply:
Sure, I can add a `log.info` in the broad exception handler with the whole exception message instead of just the type.

Reply:
Actually I take that back. `text_type(exc)` below will record the whole message from the exception, not just the type.